Framework/API/src/AlgorithmManager.cpp

```diff
@@ -253,6 +246,28 @@ void AlgorithmManagerImpl::cancelAll() {
   }
 }
 
+/// Removes all of the finished algorithms
+/// this does not lock the mutex as the locking is already assumed to be in
+/// place
+size_t AlgorithmManagerImpl::removeFinishedAlgorithms() {
+  std::vector<IAlgorithm_const_sptr> theCompletedInstances;
+  std::copy_if(
```

Comment:
Since we don't keep the elements this entire block can be replaced with

```
  size_t sizeBefore = m_managed_algs.size();

  m_managed_algs.erase(std::remove_if(
      m_managed_algs.begin(), m_managed_algs.end(), [](const auto &algorithm) {
        return (algorithm->executionState() == ExecutionState::Finished);
      }));
  size_t sizeAfter = m_managed_algs.size();

  return sizeBefore - sizeAfter;
```

Reply:
Yes, I'll change that

Reply:
Actually I'm going to revert to the previous one as the suggested code caused about 600 unit tests to segfault.

Reply:
Okay, they must be holding onto iterators that are invalidated by the erase operation, but not when we copy